integration-cli/docker_cli_build_test.go

```diff
@@ -4375,6 +4375,27 @@ func (s *DockerSuite) TestBuildTimeArgHistoryExclusions(c *check.C) {
 	}
 }
 
+// Reported in moby/moby#32744
+func (s *DockerSuite) TestBuildTimeArgHistoryExclusionsMultipleBuiltins(c *check.C) {
+	imgName := "bldargtest"
+	dockerfile := fmt.Sprintf(`FROM busybox
+		RUN echo "Testing Build Args!"`)
+	buildImage(imgName,
+		cli.WithFlags("--build-arg", "http_proxy=foo",
+			"--build-arg", "https_proxy=bar",
+		),
+		build.WithDockerfile(dockerfile),
+	).Assert(c, icmd.Success)
```

Comment:
An integration test that has to compile the entire daemon, start a daemon and run a full build is somewhat overkill for testing an array out of bounds error.

Could this be a unit test for `run()` ?